### PR TITLE
Update the Java testing documentation for 0.10 [ECR-4256]:

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
Update the testkit code examples and move them to the
main EJB repository.

### CODE REVIEW

File: src/get-started/java-binding.md [hunk 1]
```diff
@@ -1062,8 +925,8 @@ In case the service build definition needs to be configured, ensure that
 the following required metadata is present in the service artifact JAR:
 
 - Entries in the JAR manifest:
-    - "Plugin-Id": must be set to "groupId:artifactId:version", e.g.,
-    `com.exonum.example.timestamping:timestamping-demo:1.0.2`.
+    - "Plugin-Id": must be set to "1:groupId/artifactId:version", e.g.,
+    `1:com.exonum.example.timestamping/timestamping-demo:1.0.2`.
     - "Plugin-Version": must be set to the project version, e.g., `1.0.2`.
```
Comment (bullet-tooth) [inline src/get-started/java-binding.md]: Should the version be equal to a version in the plugin-id? maybe describe why such duplication is needed?

Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: Yes, it must. I'd rather not to: there is a technical reason for that that is none of our users concern.

File: src/get-started/java-binding.md [hunk 2]
```diff
@@ -682,35 +682,16 @@ Default TestKit can be instantiated with a single validator as an emulated
...
-String serviceName = "test-service";
-int serviceId = 46;
-Path artifactsDirectory = Paths.get("target");
-try (TestKit testKit = TestKit.forService(artifactId, artifactFilename,
-        serviceName, serviceId, artifactsDirectory)) {
-  // Test logic
-}
-```
+<!--codeinclude-->
+[Testkit instantiation for a single service](../../code-examples/java/exonum-java-binding/site-examples/src/test/java/com/exonum/binding/example/testing/TestkitExamplesTest.java) inside_block:testkitInstantiationForSingleService
```
Comment (slowli) [inline src/get-started/java-binding.md]: Nit / opinions: IMO, snippet titles are an overkill; the navigation on the page is cluttered already.

File: src/get-started/java-binding.md [hunk 3]
```diff
@@ -931,26 +805,9 @@ It is possible to configure the injected TestKit instance with the following ann
...
-}
-
-@Test
-void auditorTest(@Auditor @ValidatorCount(8) TestKit testKit) {
-  // Injected TestKit has an altered configuration — "auditor" as an emulated
-  // node and 8 validator nodes
-}
-```
+<!--codeinclude-->
+[TestKit per-injection re-configuration](../../code-examples/java/exonum-java-binding/site-examples/src/test/java/com/exonum/binding/example/testing/TestkitExamples2Test.java) inside_block:TestkitExamples2Test
```
Comment (slowli) [inline src/get-started/java-binding.md]: I'm getting encoding problems with this snippet locally:

> Injected TestKit has an altered configuration â€” "auditor" as an emulated

Comment (vitvakatu) [reply src/get-started/java-binding.md]: Interesting, I can't reproduce locally.

Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: Neighter can I: 
изображение 

The code has emdash at that position, but why doesn't it work on your machine?

Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: An encoding issue on Windows?

File: src/get-started/java-binding.md (hunk 2 above)
Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: But they don't seem to be added to the ToC: 
изображение

+ @vitvakatu Do you consider all captions as redundant, or only shorter ones?

**Comment by dmitry-timofeev** [comment]
> Locally, it was installed within the repository (in src/mkdocs-codeinclude-plugin), which I'd consider as unexpected behavior.

I had no specific intentions here, but there are some considerations:

> For non-editable installs, the project is built locally in a temp dir and then installed normally. *Note that if a satisfactory version of the package is already installed*, the VCS source will not overwrite it without an --upgrade flag. VCS requirements pin the *package version* (specified in the setup.py file) of the target commit, *not necessarily the commit itself.*

> The pip freeze subcommand will record the VCS requirement specifier (referencing a specific commit) if and only if the install is done using the editable option.

If installed in a venv, it is placed there (.venv/src/...).

**Comment by dmitry-timofeev** [comment]
https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs

File: src/get-started/java-binding.md (hunk 2 above)
Comment (vitvakatu) [reply src/get-started/java-binding.md]: I mean, in the following example the caption is obviously unneeded, as it duplicates the paragraph before:
изображение

File: src/get-started/java-binding.md (hunk 3 above)
Comment (slowli) [reply src/get-started/java-binding.md]: > on Windows

Yeah, probably OS-specific. The source code does have an em-dash though, so it's probably an issue somewhere in Python infrastructure.

Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: Submitted https://github.com/rnorth/mkdocs-codeinclude-plugin/pull/8 , will update our `develop` branch in #403

**Comment by dmitry-timofeev** [comment]
I wonder how do we fix these: 

    ./src/get-started/java-binding.md: 663: MD013/line-length Line length [Expected: 80; Actual: 186]

**Comment by dmitry-timofeev** [comment]
I see three possibilities:
* Add manually<sup>*</sup> `<!-- markdownlint-disable line-length -->` followed by `<!-- markdownlint-enable line-length -->` (or capture/restore pair).
* ~~Make the plugin add capture/restore pair~~ won't work as the linter works on the source Markdown file.
* Allow moving the targeting expression (block:foo) to the next line. The linter will allow a long line if it is a path.

<sup>*</sup> — One won't have to write that, just add an editor macro (e.g. Live Template in Intellij), but the reviewers will have to read that anyway.

**Comment by dmitry-timofeev** [comment]
> * Allow moving the targeting expression (block:foo) to the next line. The linter will allow a long line if it is a path.

Submitted https://github.com/rnorth/mkdocs-codeinclude-plugin/issues/9 and a patch.

**Comment by dmitry-timofeev** [comment]
With https://github.com/rnorth/mkdocs-codeinclude-plugin/pull/10 integrated into develop, it now works.

**Pull request merged**